commit 724c299c6a0e412b5679d7ebb9b3f4e00bd2aa78
Author: Marco Elver <elver@google.com>
Date:   Mon Aug 29 14:47:06 2022 +0200

    perf/hw_breakpoint: Add KUnit test for constraints accounting
    
    Add KUnit test for hw_breakpoint constraints accounting, with various
    interesting mixes of breakpoint targets (some care was taken to catch
    interesting corner cases via bug-injection).
    
    The test cannot be built as a module because it requires access to
    hw_breakpoint_slots(), which is not inlinable or exported on all
    architectures.
    
    Signed-off-by: Marco Elver <elver@google.com>
    Signed-off-by: Peter Zijlstra (Intel) <peterz@infradead.org>
    Reviewed-by: Dmitry Vyukov <dvyukov@google.com>
    Acked-by: Ian Rogers <irogers@google.com>
    Link: https://lore.kernel.org/r/20220829124719.675715-2-elver@google.com

diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index 072e4b289c13..863b51751566 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -2530,6 +2530,16 @@ config STACKINIT_KUNIT_TEST
 	  CONFIG_GCC_PLUGIN_STRUCTLEAK, CONFIG_GCC_PLUGIN_STRUCTLEAK_BYREF,
 	  or CONFIG_GCC_PLUGIN_STRUCTLEAK_BYREF_ALL.
 
+config HW_BREAKPOINT_KUNIT_TEST
+	bool "Test hw_breakpoint constraints accounting" if !KUNIT_ALL_TESTS
+	depends on HAVE_HW_BREAKPOINT
+	depends on KUNIT=y
+	default KUNIT_ALL_TESTS
+	help
+	  Tests for hw_breakpoint constraints accounting.
+
+	  If unsure, say N.
+
 config TEST_UDELAY
 	tristate "udelay test driver"
 	help